### PR TITLE
fix(multi-env): fix multi-env for local debug

### PR DESCRIPTION
**Pull request by a1exwang**
- Fix projectUpgrader. If `profile.${envName}.json` or `{envName}.userdata` do not exist, it must be a new project. It does not need to be upgraded
- Fix the valid project checking for new folder structure
- Fix local debug teamsAppId and tenantId in the new folder structure

Tested local debug.
Screenshot from 2021-08-30 15-26-08

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10728294

### CODE REVIEW

**Review by KennethBWSong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 1]
```diff
@@ -282,9 +282,31 @@ function getSettingWithUserData(jsonSelector: (jsonObject: any) => any): string
   return undefined;
 }
 
+// This is for the new folder structure for multi-env
+function getLocalSetting(jsonSelector: (jsonObject: any) => any): string | undefined {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/commonUtils.ts:286]: CLI may have the same issue.

Comment (a1exwang) [reply packages/vscode-extension/src/debug/commonUtils.ts:286]: CLI will be fixed later. Currently we focus on toolkit

**Review by swatDong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/core/tools.ts [hunk 2]
```diff
@@ -94,8 +95,13 @@ export function validateSettings(projectSettings?: ProjectSettings): string | un
...
   try {
-    const confFolderPath = path.resolve(workspacePath, `.${ConfigFolderName}`);
-    const settingsFile = path.resolve(confFolderPath, "settings.json");
+    const confFolderPath = isMultiEnvEnabled()
+      ? path.resolve(workspacePath, `.${ConfigFolderName}`, "configs")
+      : path.resolve(workspacePath, `.${ConfigFolderName}`);
+    const settingsFile = path.resolve(
+      confFolderPath,
+      isMultiEnvEnabled() ? "localSettings.json" : "settings.json"
+    );
```
Comment (dooriya) [inline packages/fx-core/src/core/tools.ts:104]: `settings.json` is used for project level setting

Comment (a1exwang) [reply packages/fx-core/src/core/tools.ts:105]: fixed

**Review by dooriya — APPROVED** [review]

**Review by qinezh — APPROVED** [review]

**Pull request merged**